src/views/dataset/MultiSourceConfiguration.vue

```diff
@@ -260,6 +260,122 @@ enum Sources {
+  if (ms < 1000) return `${Math.round(ms)} ms`;
+  const s = ms / 1000;
+  if (s < 60) return `${trimFloat(s)} s`;
+  const m = s / 60;
+  if (m < 60) return `${trimFloat(m)} min`;
+  const h = m / 60;
+  if (h < 24) return `${trimFloat(h)} h`;
+  const d = h / 24;
+  return `${trimFloat(d)} d`;
+}
+
+function formatDistanceShort(um: number): string {
+  if (!isFinite(um)) return "";
+  // Pick µm by default; switch to mm if large
+  if (Math.abs(um) >= 1000) return `${trimFloat(um / 1000)} mm`;
+  if (Math.abs(um) >= 1) return `${trimFloat(um)} µm`;
+  // go to nm if very small
+  return `${trimFloat(um * 1000)} nm`;
+}
+
+function trimFloat(n: number): string {
+  // Up to 3 sig-ish digits without trailing zeros
+  const s =
+    Math.abs(n) >= 100
+      ? n.toFixed(0)
+      : Math.abs(n) >= 10
+        ? n.toFixed(1)
+        : Math.abs(n) >= 1
+          ? n.toFixed(2)
+          : Math.abs(n) >= 0.1
+            ? n.toFixed(3)
+            : Math.abs(n) >= 0.01
+              ? n.toFixed(4)
+              : n.toFixed(5);
+  return s.replace(/(?:\.0+|(\.\d*?[1-9])0+)$/, "$1");
+}
+
+// ---- Core extraction from ND2 metadata ----
+function getTimeLabels(meta: Nd2LikeMeta): string[] | null {
+  const exps = meta.nd2_experiment ?? [];
+  const time = exps.find((e) => e.type === "TimeLoop") as
+    | Extract<Nd2ExperimentEntry, { type: "TimeLoop" }>
+    | undefined;
+  if (!time) return null;
+
+  const count = Math.max(0, time.count | 0);
+  const periodMs = Math.max(0, time.parameters.periodMs || 0);
+
+  const labels: string[] = [];
+  for (let i = 0; i < count; i++) {
+    labels.push(formatDurationShort(i * periodMs));
+  }
+  return labels;
+}
+
+function getZLabels(meta: Nd2LikeMeta): string[] | null {
+  const exps = meta.nd2_experiment ?? [];
+  const z = exps.find((e) => e.type === "ZStackLoop") as
+    | Extract<Nd2ExperimentEntry, { type: "ZStackLoop" }>
+    | undefined;
+  if (!z) return null;
+
+  const count = Math.max(0, z.count | 0);
+  const stepUm = Number(z.parameters.stepUm ?? 0) || 0;
+
+  // If homeIndex exists, center around it; else start at 0
+  const hasHome = Number.isFinite(z.parameters.homeIndex);
+  const home = hasHome ? (z.parameters.homeIndex as number) : 0;
+
+  const valuesUm: number[] = [];
+  for (let i = 0; i < count; i++) {
+    const delta = hasHome ? (i - home) * stepUm : i * stepUm;
+    valuesUm.push(delta);
+  }
+
+  return valuesUm.map((v) => formatDistanceShort(v));
+}
+
+function getXYLabels(meta: Nd2LikeMeta): string[] | null {
+  const exps = meta.nd2_experiment ?? [];
+  const xy = exps.find((e) => e.type === "XYPosLoop") as
+    | Extract<Nd2ExperimentEntry, { type: "XYPosLoop" }>
+    | undefined;
+  if (!xy) return null;
+
+  const points = xy.parameters.points;
+  if (!points || points.length === 0) return null;
+
+  return points.map((point) => {
+    const [x, y] = point.stagePositionUm;
+    return `${trimFloat(x)}, ${trimFloat(y)}`;
+  });
+}
+
```

Comment:
These should go in a separate file (like `ND2Helpers.ts`)  
MultiSourceConfiguration.vue is already 1600 lines long ^^'